Initial device registration policy - Preview docs (#12257)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/azureadregistrationpolicy.md

---
title: "azureADRegistrationPolicy resource type"

> description: "Properties and relationships of the azureADRegistrationPolicy resource"

Reviewer: ```suggestion
description: "Represents the policy scope of an Active Directory tenant that controls device registration using Azure AD registration."
```

author: "spunukol"
localization_priority: Normal
ms.prod: "directory-management"
doc_type: resourcePageType
---

# azureADRegistrationPolicy resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents the policy scope of the Azure Active Directory tenant that controls the ability for users and groups to register device identities to your organization using Azure AD registered. Only the read and update operations are supported on this resource; create and delete are not supported.

Reviewer: ```suggestion
Represents the policy scope of the Azure Active Directory tenant that controls the ability for users and groups to register device identities to your organization using Azure AD registered.
```

> 

Reviewer: Should be a section titled Methods and in that section should be a table listing the Get and Update methods with links to the method pages. See other published resource pages for examples.

Method | Return Type
-- | --

## Properties
|Property|Type|Description|
|:---|:---|:---|

> |allowedGroups|String collection|Group IDs in scope of policy. Required when used with the `selected` value in the `appliesTo` property.|

Reviewer: ```suggestion
|allowedGroups|String collection| The group identifiers that are in the scope of the policy. Required when used with the `selected` value in the `appliesTo` property.|
```

> |allowedUsers|String collection|User IDs in scope of policy. Required when used with the `selected` value in the `appliesTo` property.|

Reviewer: ```suggestion
|allowedUsers|String collection| The user identifiers that are in the scope of the policy. Required when used with the `selected` value in the `appliesTo` property.|
```

> |appliesTo|policyScope|Specifies whether to block, allow or fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. Required.|

Reviewer: ```suggestion
|appliesTo|policyScope|Specifies whether to block or allow fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. Required.|
```

|isAdminConfigurable|Boolean|Specifies whether this `policyScope` is configurable by the admin. Read-only.|

> ## Special considerations when using `appliesTo`

Reviewer: This should not be a new section. The information should just be a continuation of the information in the properties section.

Author: JoFlores has fixed up all of the PR while I was OOF. pls review

---
title: "azureADRegistrationPolicy resource type"
description: "Represents the policy scope of an Active Directory tenant that controls device registration using Azure AD registration."
author: "spunukol"
localization_priority: Normal
ms.prod: "directory-management"
doc_type: resourcePageType
---
# azureADRegistrationPolicy resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Represents the policy scope of the Azure Active Directory tenant that controls the ability for users and groups to register device identities to your organization using Azure AD registered.

> ## Methods
> 
> | Method | Return Type | Description |
> | --- | --- | --- |
> | [Get deviceRegistrationPolicy](../api/deviceregistrationpolicy-get.md) | [deviceRegistrationPolicy resource type](deviceregistrationpolicy.md) | Read the properties of a **deviceRegistrationPolicy** object. |
> | [Update deviceRegistrationPolicy](../api/deviceregistrationpolicy-update.md) | [deviceRegistrationPolicy resource type](deviceregistrationpolicy.md) | Update the properties of a **deviceRegistrationPolicy** object. |
> 

Reviewer 2: This resource looks like a complex type. Methods only apply to the main entity type, **deviceRegistrationPolicy** resource.
```suggestion
```

## Properties

|Property|Type|Description|
|:---|:---|:---|

> |allowedGroups|String collection| The group identifiers that are in the scope of the policy. Required when used with the `selected` value in the `appliesTo` property. |
> |allowedUsers|String collection| The user identifiers that are in the scope of the policy. Required when used with the `selected` value in the `appliesTo` property. |
> |appliesTo|policyScope| Specifies whether to block or allow fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. Required. |
> |isAdminConfigurable|Boolean|Specifies whether this `policyScope` is configurable by the admin. Read-only.|

Reviewer 2: A few observations:
- appliesTo property returns 0,1 ... string values not `none`, `all`, `selected`.
  - Ideally, customers always expect the API to return the string name, not its int representation. Should changes be made to the schema before the docs are published?
- isAdminConfigurable property is updatable.
- None of the properties is required.

```suggestion
|allowedGroups|String collection| The identifiers of the groups that are in the scope of the policy. Required when the **appliesTo** property is set to `selected`. |
|allowedUsers|String collection| The identifiers of users that are in the scope of the policy. Required when the **appliesTo** property is set to `selected`. |
|appliesTo|policyScope| Specifies whether to block or allow fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. The default value is `0`. The default value is `0`. When set to `2`, at least one user or group identifier must be specified in either **allowedUsers** or **allowedGroups**.  Setting this property to `0` or `1` removes all identifiers in both **allowedUsers** and **allowedGroups**. |
|isAdminConfigurable|Boolean|Specifies whether this policy scope is configurable by the admin. The default value is `false`. |
```

* The default value is **`all`** at time of policy creation.
* You must have at least one **`allowedGroup`** or **`allowedUser`** when using **`selected`**.
* The value of **`all`** or **`none`** will remove all values in **`allowedGroups`** and **`allowedUsers`**.

> * The values defaults to **`all`** when **`isAdminConfigurable`** is set to **`false`**. This is when an admin has enabled Intune(MEM) application to manage devices in Azure Active Directory.

Reviewer 2: - Deleting to merge this into the property descriptions.
- Observation: **isAdminConfiguration** & **appliesTo** are not strictly dependent as the last statement declared. I'm able to set to (false & 1 or false & 0) and (true & 1 or true & 0). Note: appliesTo returns 0, 1, 2, not `none`, `all`, `selected`.

Author: thx
